src/Workspaces/VisualBasic/Portable/Formatting/Engine/Trivia/TriviaDataFactory.Analyzer.vb

```diff
@@ -71,19 +72,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Formatting
                 result.Tab = 0
             End Sub
 
-            Private Shared Sub AnalyzeLineBreak(trivia As SyntaxTrivia, ByRef result As AnalysisResult)
-                ' if there was any space before line break, then we have trailing spaces
-                If result.Space > 0 OrElse result.Tab > 0 Then
+            Private Shared Sub AnalyzeLineBreak(previousTrivia As SyntaxTrivia, trivia As SyntaxTrivia, ByRef result As AnalysisResult)
+                ' if there was any space immediately before line break, then we have trailing spaces
+                If previousTrivia.Kind = SyntaxKind.WhitespaceTrivia AndAlso previousTrivia.Width > 0 Then
                     result.HasTrailingSpace = True
+                    result.HasTabAfterSpace = False
+                    result.Space = 0
+                    result.Tab = 0
+                    result.TreatAsElastic = result.TreatAsElastic Or trivia.IsElastic()
```

Comment:
❔ What happens if these three lines are left in their original location?

Reply:
@sharwell 4 tests fail for example ColonToken_If with whitepace NOT removed
```
Expected: ···        If True Then\r\n        End If\r\n    End Sub\r\nEnd Class
   Actual:   ···        If True Then \r\n        End If\r\n    End Sub\r\nEnd Class
```